test(lsp): add simple tests for `textDocument/didOpen|didChange|didClose` and `textDocument/codeAction`

Sysix opened the pull request:
> This PR adds comprehensive tests for the LSP (Language Server Protocol) server's text document lifecycle notifications (textDocument/didOpen, textDocument/didChange, textDocument/didClose) and code action functionality (textDocument/codeAction). The changes include both unit-level tests in worker.rs and integration-level tests in tests.rs.

Sysix commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16404?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#16404** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16404?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16404?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#16361** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16361?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#16360** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16360?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=16404).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_language_server/src/tests.rs:
```diff
@@ -851,4 +916,67 @@ mod test_suite {
+        assert!(response.id() == &Id::Number(3));
+        assert!(response.result().is_some_and(|result| *result == Value::Null));
+
+        server.shutdown_with_watchers(4).await;
```
yes, but no this is a bug :) will be fixed in upstream PR

camc314 commented:
### Merge activity

* **Dec 2, 6:18 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 2, 6:18 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
test(lsp): add simple tests for `textDocument/didOpen|didChange|didClose` and `textDocument/codeAction` (#16404)

> This PR adds comprehensive tests for the LSP (Language Server Protocol) server's text document lifecycle notifications (textDocument/didOpen, textDocument/didChange, textDocument/didClose) and code action functionality (textDocument/codeAction). The changes include both unit-level tests in worker.rs and integration-level tests in tests.rs.
